cmd/seed.go

```diff
@@ -0,0 +1,196 @@
+package cmd
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+
+	"github.com/MakeNowJust/heredoc"
+	shieldv1beta1 "github.com/raystack/shield/proto/v1beta1"
+	cli "github.com/spf13/cobra"
+	"google.golang.org/grpc/metadata"
+)
+
+var (
+	//go:embed seed/permissions.json
+	mockCustomPermissions []byte
+	//go:embed seed/roles.json
+	mockCustomRoles []byte
+	//go:embed seed/user.json
+	mockHumanUser []byte
+	//go:embed seed/organization.json
+	mockOrganization []byte
+	//go:embed seed/project.json
+	mockProject []byte
+	//go:embed seed/resource.json
+	mockResource []byte
+
+	orgAdminEmail  = "sampleAdmin@raystack.org"
+	identityHeader = "X-Shield-Email"
+)
+
+func SeedCommand(cliConfig *Config) *cli.Command {
+	cmd := &cli.Command{
+		Use:   "seed",
+		Short: "Seed the database with initial data",
+		Args:  cli.NoArgs,
+		Long:  "This command can be used to create an organization structure with predefined groups, projects, and resources. It bootstarps these data in the Shield db, making it easier to get started.",
+		Example: heredoc.Doc(`
+			$ shield seed
+		`),
+		Annotations: map[string]string{
+			"group": "core",
+		},
+
+		RunE: func(cmd *cli.Command, args []string) error {
+			adminClient, cancel, err := createAdminClient(cmd.Context(), cliConfig.Host)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			// create custom roles and permissions for the sample data
+			if err := createCustomRolesAndPermissions(adminClient); err != nil {
+				return fmt.Errorf("failed to create custom permissions: %w", err)
+			}
+
+			client, cancel, err := createClient(cmd.Context(), cliConfig.Host)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			if err := bootstrapData(client); err != nil {
+				return fmt.Errorf("failed to bootstrap data: %w", err)
+			}
+
+			fmt.Println("initialized sample data in shield successfully")
+			return nil
+		},
+	}
+
+	bindFlagsFromClientConfig(cmd)
+
+	return cmd
+}
+
+// create compute/order namespace and custom roles and permissions
+func createCustomRolesAndPermissions(client shieldv1beta1.AdminServiceClient) error {
+	ctx := metadata.NewOutgoingContext(context.Background(), metadata.New(map[string]string{
+		identityHeader: orgAdminEmail,
+	}))
+
+	var permissionBodies []*shieldv1beta1.PermissionRequestBody
+	if err := json.Unmarshal(mockCustomPermissions, &permissionBodies); err != nil {
+		fmt.Println("Error unmarshaling JSON :", err)
+		return err
+	}
+
+	// create custom permission
+	if _, err := client.CreatePermission(ctx, &shieldv1beta1.CreatePermissionRequest{
+		Bodies: permissionBodies,
+	}); err != nil {
+		return fmt.Errorf("failed to create custom permission: %w", err)
+	}
+
+	fmt.Printf("created custom permissions for compute order service: %s, %s, %s, %s, %s\n", "create", "get", "update", "delete", "list")
```

Comment:
Permissions are read from json, hardcoding this log can go out of sync. Should we also print this based on json?